components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java

```diff
@@ -95,7 +108,10 @@ public StorageFacetImpl(final BlobStoreManager blobStoreManager,
                           final @Named(ComponentDatabase.NAME) Provider<DatabaseInstance> databaseInstanceProvider,
                           final BucketEntityAdapter bucketEntityAdapter,
                           final ComponentEntityAdapter componentEntityAdapter,
-                          final AssetEntityAdapter assetEntityAdapter)
+                          final AssetEntityAdapter assetEntityAdapter,
+                          final ClientInfoProvider clientInfoProvider,
+                          final Map<String, ContentValidator> contentValidators,
+                          final DefaultContentValidator defaultContentValidator)
```

Comment:
Not entirely convinced by that argument ;) but if that's the case then I'd say this logic should be extracted out to a component or service to make it easier to mock/test (ditto for the logic in ViewServlet/HttpResponseSender)